.github/workflows/snap-testing.yml

```diff
@@ -56,10 +77,21 @@ jobs:
       - name: Checkout the local actions again
         uses: actions/checkout@v2
 
+      - name: Build snap
+        if: ${{ github.event_name == 'workflow_dispatch' && inputs.localBuild }}
+        uses: ./build
+        id: build
+        with:
+          repo: ${{matrix.repo}}
+
+      - name: Checkout the local actions
+        uses: actions/checkout@v2
```

Comment:
There is a step named "Checkout the local actions again". I believe that is from before when there was a build stage.
You can remove this and move "Build snap" to before that existing, second checkout.